include/depthai/depthai_wrapper.hpp

```diff
@@ -0,0 +1,70 @@
+#include "depthai/device.hpp"
+#include "nlohmann/json.hpp"
+#include <opencv2/core/core.hpp>
+#include <opencv2/opencv.hpp>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace DepthAI {
+
+using CV_mat_ptr = std::shared_ptr<cv::Mat>; // shared ptr for cv::Mat
+using PacketsTuple = std::tuple<
+    std::list<std::shared_ptr<NNetPacket>>,
+    std::list<std::shared_ptr<HostDataPacket>>>; // tuple of data packets on which Depthai is
+// publishing the streams
+
+/**  DepthAI:
+ *   This is a wrapper on the host side of the device which 
+ *   helps in initializing device, creating pipeline
+ *   and fetching the frames as cv::Mat. 
+ */
+class DepthAI : public Device {
+
+public:
+    /**  Constructor:
+     *   initializing device and creating pipeline
+     *  @param usb_device (std::string): Provide the path of the device
+     *  or pass empty string to choose the default device.
+     *  @param config_file (std::string): Provides the json file which is
+     *  used by the Depthai to configure the device
+     *  @param usb2_mode (bool): set to true to connect over usb2
+     */
+    DepthAI(const std::string& usb_device, const std::string& config_file, bool usb2_mode = false);
+
+    /* API to stream output frames from device.
+     * AI models results and metadata is WIP
+     * @param output_streams (std::unordered_map<std::string, CV_mat_ptr>): output_streams is a placeholder 
```

Comment:
Just wondering, are all output streams `CV_mat_ptr`s? What is the output format of the AI detections? I can't imagine them to be `CV_mat_ptr` right? Also, I don't understand why this function is called get frames, but the output is streams? Do you get 1 frame of each of the streams?

Reply:
No. you are right. currently I am doing it only for image streams. I haven't added AI detection and metadata. which are not cv::Mat. I will be adding them in the next PR.

Reply:
What about my second question?

Reply:
Yes you get one frame of all the streams you have specified